module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1227,4 +1289,97 @@ public function __call($methodName, $params)
             'Cannot call method: ' . $this->getDriverClass() . '::' . $methodName
         );
     }
+
+    /**
+     * Get data for an ILS method from shared or session cache
+     *
+     * @param array $cacheSettings Cache settings
+     *
+     * @return ?array
+     */
+    protected function getCachedData(array $cacheSettings): ?array
+    {
+        $cacheKey = $cacheSettings['key'];
+        if ('shared' === $cacheSettings['storage']) {
+            return $this->getSharedCachedData($cacheKey);
+        }
+        if ($this->sessionCache && ($entry = $this->sessionCache[$cacheKey] ?? null)) {
+            if (time() - $entry['ts'] <= $cacheSettings['lifeTime']) {
+                return $entry['payload'];
+            }
+            unset($this->sessionCache[$cacheKey]);
+        }
+        return null;
+    }
+
+    /**
+     * Put data for an ILS method to shared or session cache.
+     *
+     * @param array $cacheSettings Cache settings
+     * @param array $data          Data to cache
+     *
+     * @return void
+     */
+    protected function putCachedData(array $cacheSettings, array $data): void
+    {
+        $cacheKey = $cacheSettings['key'];
+        if ('shared' === $cacheSettings['storage']) {
+            $this->putSharedCachedData($cacheKey, $data);
```

Comment:
Shouldn't you be accounting for the lifetime here? Otherwise I think the default 30 seconds will be used, unless I'm missing something!

Reply:
Yikes! Thanks, for catching that. The call was missing the third parameter.